app/src/main/kotlin/ExtendedStatelessSolver.kt

```diff
@@ -0,0 +1,40 @@
+import kotlin.math.*
+
+class ExtendedStatelessSolver<TState, TAction>(
+        mdp: MDP<TState, TAction>,
+        simulationDepthLimit: Int,
+        explorationConstant: Double,
+        rewardDiscountFactor: Double,
+        verbose: Boolean
+) : StatelessSolver<TState, TAction>(mdp, simulationDepthLimit, explorationConstant, rewardDiscountFactor, verbose) {
+
+    val rewardHistory = mutableListOf<Double>()
+
+    override fun constructTree(iterations: Int) {
+        for (i in 0..iterations) {
+            iterateStep()
+
+            val bestChild = _root.getChildren().maxByOrNull { c -> calculateUCT(c)}
```

Comment:
@JunTaoLuo is this convention to name private variables with _xyz. It goes against [Java naming convention](https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html)

Reply:
Ah yes, this is a c#ism that I'll need to fix.